src/System.Net.Sockets.Legacy/tests/FunctionalTests/DualModeSocketTest.cs

```diff
@@ -1007,7 +1006,7 @@ private void Accept_Helper(IPAddress listenOn, IPAddress connectTo)
                 SocketClient client = new SocketClient(serverSocket, connectTo, port);
                 Socket clientSocket = serverSocket.Accept();
                 Assert.True(clientSocket.Connected);
-                Assert.True(clientSocket.DualMode);
+                Assert.True((listenOn != IPAddress.IPv6Any && !listenOn.IsIPv4MappedToIPv6) || clientSocket.DualMode);
```

Comment:
I think of tests as our functional spec. While the changes look nicer this way (more compact) it would be clearer for us and for our customers if we had two tests or use the API above to clearly point out that the behavior is different by design on different platforms.